frame/support/src/inherent.rs

```diff
@@ -97,6 +97,41 @@ macro_rules! impl_outer_inherent {
 					)*
 				}
 
+				$(
+					match $module::is_inherent_required(self) {
```

Comment:
I'm not really happy that we iterate over all inherents for each module, but I also don't know a better solution.

Reply:
It shouldn't impact existing runtime performance, as we only iterate when `is_inherent_required` returns `Some`, which for majority of the time will not be the case (even for `anyupgrade`, `Some` is only returned on particular block numbers).

Reply:
Good point.